### PR TITLE
fix: make sure x-graphql-event-stream url is domain relative

### LINKED ISSUE #1143: x-graphql-event-stream is invalid when graphqlRoute is "/"
when `graphqlRoute` is "/", `x-graphql-event-stream` become `//stream` and that's understood as a protocol relative uri by browsers, so they try to connect to https://stream/

one easy option would be to remove trailling "/" from `graphqlRoute` when building the `x-graphql-event-stream` header

I'd like it if `graphqlRoute: ""` was not mapped to `graphqlRoute: "/graphql"` but only when `null` or `undefined` but this would break backward compatibility

### PR DESCRIPTION
**Pull request by mathroc**
closes #1143

### CODE REVIEW

File: src/postgraphile/http/createPostGraphileHttpRequestHandler.ts [hunk 1]
```diff
@@ -387,7 +387,9 @@ export default function createPostGraphileHttpRequestHandler(
           /<\/head>/,
           `  <script>window.POSTGRAPHILE_CONFIG=${safeJSONStringify({
             graphqlUrl: `${externalUrlBase}${graphqlRoute}`,
-            streamUrl: watchPg ? `${externalUrlBase}${graphqlRoute}/stream` : null,
+            streamUrl: watchPg
+              ? `${externalUrlBase}${graphqlRoute}`.replace(/\/*$/, '') + `/stream`
```
Comment (benjie) [inline src/postgraphile/http/createPostGraphileHttpRequestHandler.ts]: I’m not at my computer; but would this be equivalent? If so, I’d much prefer it for DRY, simplicity and performance reasons 👍

```suggestion
              ? `${externalUrlBase}${streamRoute}`
```

File: src/postgraphile/http/createPostGraphileHttpRequestHandler.ts [hunk 2]
```diff
@@ -562,7 +564,10 @@ export default function createPostGraphileHttpRequestHandler(
     if (watchPg) {
       // Inform GraphiQL and other clients that they can subscribe to events
       // (such as the schema being updated) at the following URL
-      res.setHeader('X-GraphQL-Event-Stream', `${externalUrlBase}${graphqlRoute}/stream`);
+      res.setHeader(
+        'X-GraphQL-Event-Stream',
+        `${externalUrlBase}${graphqlRoute}`.replace(/\/*$/, '') + `/stream`,
```
Comment (benjie) [inline src/postgraphile/http/createPostGraphileHttpRequestHandler.ts]: As above

File: src/postgraphile/http/createPostGraphileHttpRequestHandler.ts (hunk 1 above)
Comment (mathroc) [reply src/postgraphile/http/createPostGraphileHttpRequestHandler.ts]: yes, I didn't notice before that `externalUrlBase` can't end with à `/`

**Comment by mathroc** [comment]
should be fine now

**Pull request merged**